src/mvPlotting.cpp

```diff
@@ -458,40 +392,34 @@ DearPyGui::draw_plot(ImDrawList* drawlist, mvAppItem& item, mvPlotConfig& config
 		if (config._useColorMap)
 			ImPlot::PopColormap();
 
-		config._queried = ImPlot::IsPlotQueried();
-
-		if (config._queried)
-		{
-			ImPlotLimits area = ImPlot::GetPlotQuery();
-			config._queryArea[0] = area.X.Min;
-			config._queryArea[1] = area.X.Max;
-			config._queryArea[2] = area.Y.Min;
-			config._queryArea[3] = area.Y.Max;
-		}
-
-		if (item.config.callback != nullptr && config._queried)
+		if (item.config.callback != nullptr && config.rects.size() > 0)
 		{
-
-			if (item.config.alias.empty())
-				mvSubmitCallback([=, &item]() {
-				PyObject* area = PyTuple_New(4);
-				PyTuple_SetItem(area, 0, PyFloat_FromDouble(config._queryArea[0]));
-				PyTuple_SetItem(area, 1, PyFloat_FromDouble(config._queryArea[1]));
-				PyTuple_SetItem(area, 2, PyFloat_FromDouble(config._queryArea[2]));
-				PyTuple_SetItem(area, 3, PyFloat_FromDouble(config._queryArea[3]));
-				mvAddCallback(item.config.callback, item.uuid, area, item.config.user_data);
-					});
-			else
+			for(auto rect : config.rects) {
+				const std::string sender = item.config.alias.empty() ? std::to_string(item.uuid) : item.config.alias;
```

Comment:
The `sender` argument in the callback must be either a numeric UUID or a string alias (tag). By converting UUID to a string, this code makes it impossible for the callback to pass `sender` directly to DPG API (e.g. try to call `dpg.does_item_exist(sender)` on a plot without `tag` - it will return `False`).

For now, let's do it via a copy-paste of `mvSubmitCallback` just like the rest of DPG code does. In future, I'm going to push a PR that removes all that `mvSubmitCallback` duplication and wraps these pairs into a single call receiving both UUID and alias.